supertokens_python/recipe/session/recipe_implementation.py

```diff
@@ -46,10 +48,107 @@
 if TYPE_CHECKING:
     from typing import List, Union
     from supertokens_python import AppInfo
-    from supertokens_python.querier import Querier
 
-from .constants import JWKCacheMaxAgeInMs, JWKRequestCooldownInMs
 from .interfaces import SessionContainer
+from supertokens_python.querier import Querier
+
+
+class GetJWKSResult:
+    def __init__(self, keys: List[PyJWK], last_fetched: int):
+        self.keys = keys
+        self.last_fetched = last_fetched
+
+
+from typing_extensions import TypedDict
+import threading
+from os import environ
+
+
+class JWKSConfigType(TypedDict):
+    cache_max_age: int
+    refresh_rate_limit: int
+
+
+JWKSConfig: JWKSConfigType = {
+    "cache_max_age": 6000,
+    "refresh_rate_limit": 500,
+}
+
+jwks_cache: Optional[GetJWKSResult] = None
+mutex = threading.RLock()
```

Comment:
Does this behave like a read write lock? How do you stop reads to the cache when you are refreshing they keys? (while allowing parallel reads)

Reply:
I would add tests that use the custom RWMutex to make sure it works correctly

Reply:
Added a test for writes `test_rw_mutex_writes`. 
We already had a `test_that_locking_for_jwks_cache_works` btw